data/actions/actionsRouter.js

```diff
@@ -0,0 +1,88 @@
+const express = require("express");
+const db = require("../helpers/actionModel");
+const router = express.Router();
+router.use(express.json());
+
+
+router.get("/", (req, res) => {
+ db.get()
+ .then(actions => {
+   console.log(actions)
+   res.status(200).json(actions);
+ })
+ .catch(err => {
+   res.status(500).json({error: "Cant do that !"}) })
+ });
+ 
+ router.get("/:id",validateActionsId, (req, res) => {
+   const id = req.params.id;
```

Comment:
you are already getting your id in your first middleware (ln 74). Try to look into how to PASS data / saved variables through middleware. 

I found this video interesting! (res.locals): https://www.youtube.com/watch?v=zPYmM9K8-g8